react_juce/core/TextInputView.cpp

```diff
@@ -0,0 +1,207 @@
+            dirty = false;
+        }
+    }
+
+    //==============================================================================
+
+    TextInputView::TextInputView()
+        : textInput(props)
+    {
+        addAndMakeVisible(textInput);
+        textInput.addListener(&textInput);
+        textInput.setPopupMenuEnabled(false);
+    }
+
+    void TextInputView::setProperty(const juce::Identifier &name, const juce::var &value)
+    {
+        View::setProperty(name, value);
+        if (name == valueProp)
+        {
+            if (!value.isString())
+                throw std::invalid_argument("Invalid prop value. Prop \'value\' must be a string.");
+            textInput.setControlled(true);
+            textInput.setControlledValue(value);
+        }
+        if (name == placeholderProp)
+        {
+            if (!value.isString())
+                throw std::invalid_argument("Invalid prop value. Prop \'placeholder\' must be a string.");
+            textInput.setPlaceholderText(value);
+        }
+        if (name == placeholderColorProp)
+        {
+            if (!value.isString())
+                throw std::invalid_argument("Invalid prop value. Prop \'placeholder-color\' must be a color string.");
+            juce::String hexPlaceholderColor = value;
+            juce::Colour placeholderColor = juce::Colour::fromString(hexPlaceholderColor);
+            textInput.setPlaceholderColour(placeholderColor);
+        }
+        if (name == maxlengthProp)
+        {
+            if (!value.isDouble())
+              throw std::invalid_argument("Invalid prop value. Prop \'maxlength\' must be a number.");
+            textInput.setMaxLength(value);
+        }
+        if (name == readonly)
+        {
+            textInput.setReadOnly(value);
+        }
+
+        textInput.applyFontToAllText(getFont());
+
+        juce::String hexColor = props.getWithDefault(colorProp, "ff000000");
+        juce::Colour colour = juce::Colour::fromString(hexColor);
+        textInput.applyColourToAllText(colour);
+
+        const int just = props.getWithDefault(justificationProp, 1);
+        textInput.setJustification(just);
+
+        if (isTextEditorColorProp(name))
+            setTextEditorColorProp(name, value);
+    }
+
+    void TextInputView::resized()
+    {
+        View::resized();
+        textInput.setBounds(getLocalBounds());
+    }
+
+    juce::Font TextInputView::getFont()
+    {
+        return TextView::getFont(props);
+    }
+
+    bool TextInputView::isTextEditorColorProp(const juce::Identifier &textEditorColorProp)
+    {
+        auto it = textEditorColourIdsByProp.find(textEditorColorProp);
+        bool found = (it != textEditorColourIdsByProp.end());
+        return found;
+    }
+
+    void TextInputView::setTextEditorColorProp(const juce::Identifier &textEditorColorProp, const juce::var &value)
+    {
+        if (!value.isString())
+        {
+            std::string propName = textEditorColorProp.toString().toStdString();
```

Comment:
Most of these inline vars in `setTextEditorColorProp` can be `const`

Reply:
This is much nicer though! Good stuff.

Reply:
Thanks!
I added const in 973a63c.